README.rst

```diff
@@ -1,133 +1,190 @@
+submitted as a job to a Flink job manager for demonstration purposes, but can
+also be invoked interactively.
 
-Kafka->CrateDB job
-------------------
+- The ``SimpleJdbcSinkJob`` demonstrates a basic example which inserts a few
+  records into CrateDB using Flink. It outlines how to use the fundamental
+  Flink ``JdbcSink`` API, and how to adjust the corresponding
+  ``JdbcExecutionOptions`` and ``JdbcConnectionOptions``.
 
-Build job Jar file.
+- The ``SimpleTableApiJob`` demonstrates how to use the Flink Table API and
+  the Flink DataStream API.
 
-.. code:: console
+  The `Flink Table API`_ is a language-integrated query API for Java, Scala, and
+  Python that allows the composition of queries from relational operators such as
+  selection, filter, and join in a very intuitive way, and offers a unified
+  interface for both stream and batch processing.
 
-    $ ./gradlew build
+  The `Flink DataStream API`_ offers the primitives of stream processing (namely
+  time, state, and dataflow management) in a relatively low-level imperative
+  programming API. The Table API abstracts away many internals and provides a
+  structured and declarative API.
 
-Submit to Flink.
+  Both APIs can work with bounded and unbounded streams. Bounded streams need to
+  be managed when processing historical data. Unbounded streams occur in
+  real-time processing scenarios that might be initialized with historical data
+  first.
 
-.. code:: console
+- The ``TaxiRidesStreamingJob`` subscribes to an `Apache Kafka`_ topic as a data
+  source, and stores received data into `CrateDB`_ as a data sink. The data stream
+  is represented by records from the venerable `NYC Yellow Taxi Trips`_ dataset.
 
-    make submit
 
-In-process jobs
----------------
-
-At ``io.crate.flink.demo``, there are two Flink example jobs/programs which can
-be launched interactively without needing to submit them. ``SimpleJdbcSinkJob``
-demonstrates how to insert records into CrateDB using the official
-``org.postgresql.Driver`` JDBC driver.
-
-.. code:: console
+*****
+Usage
+*****
 
-    make run JOB=SimpleJdbcSinkJob
+Prerequisites
+=============
 
-``SimpleTableApiJob`` demonstrates how to leverage the virtual
-``TableEnvironment`` subsystem.
+Acquire and build the source code.
 
 .. code:: console
 
-    make run JOB=SimpleTableApiJob
-
+    git clone https://github.com/crate/cratedb-flink-jobs
+    cd cratedb-flink-jobs
+    make build
 
-Flink Job Configurations
-========================
-
-These settings for the sample Flink job.
-
-**kafka.servers**
-  | *required:* true
-
-  The comma separated list of Kafka brokers.
-
-**kafka.group.id**
-  | *default:* default
-
-  The Kafka consumer group ID.
+``TaxiRidesStreamingJob``
+=========================
 
-**kafka.topic**
-  | *required:* true
+- ``make test`` will probe the job using ``flink info``.
+- ``make submit`` will submit the job using ``flink run`` to a Flink job
+  manager at ``localhost:8081`` .
```

Comment:
Currently, it will raise this error when invoking `make submit` without having a Flink cluster running.
```
docker: Error response from daemon: network scada-demo not found.
make: *** [Makefile:9: submit] Error 125
```
```
$ echo $?
2
```

Reply:
But it  will also raise error if you have a Flink cluster but no CrateDB and/or no Kafka, right?

Reply:
### W/o CrateDB
Exception for `make submit` when CrateDB has been stopped:
```java
Caused by: org.apache.flink.runtime.client.JobExecutionException: Job execution failed.
Caused by: org.apache.flink.runtime.JobException: Recovery is suppressed by NoRestartBackoffTimeStrategy
Caused by: java.io.IOException: unable to open JDBC writer
Caused by: org.postgresql.util.PSQLException: The connection attempt failed.
Caused by: java.net.UnknownHostException: cratedb
```

### W/o Flink task manager
Exception for `make submit` when Flink task manager has been stopped:
```java
Caused by: org.apache.flink.runtime.jobmanager.scheduler.NoResourceAvailableException: Could not acquire the minimum required resources.
```

### W/o Kafka broker
Exception for `make submit` when Kafka broker has been stopped:
```java
Caused by: org.apache.kafka.common.KafkaException: Failed to construct kafka consumer
Caused by: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers
```

Reply:
That's why imho, It would be better to add some comment and point out to the cratedb-examples repo and kafka stack, wdyt?

Reply:
That's awesome! thx a lot!